src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -845,47 +855,119 @@ QgsExpressionContext QgsLayoutItemLegend::createExpressionContext() const
 
   context.appendScope( scope );
 
-  return context;
+  if ( replace )
+  {
+    mExpContext.~QgsExpressionContext();
+    QList<QgsExpressionContextScope *> scopes;
```

Comment:
```suggestion
    const QList<QgsExpressionContextScope *> scopes;
```

Make sure range based for loops are always called on const containers (or they will detach which is an expensive operation)

Reply:
should I also const [this line?](https://github.com/roya0045/QGIS/blob/a8d3de8bb7b5ef54e9b5a1975ac35351e7d109a2/src/core/qgsvectorlayerfeaturecounter.cpp#L70)

I assume that only returned const objets could be const and that normal things needed to be casted as const. I guess the travis compilation will tell if those items cannot be const.